test/unittests/utils/test_xml_utils.py

```diff
@@ -1,56 +1,67 @@
 
-# ruff: noqa: PT009 (pytest-unittest-assertion) (remove this line when pytest is used instead of unittest)
+
+@pytest.fixture()
+def data_systematic_unclean() -> Any:
+    return etree.parse("testdata/xml-data/test-data-systematic.xml")
+
+
+@pytest.fixture()
+def data_systematic_cleaned() -> etree._Element:
+    return parse_and_clean_xml_file("testdata/xml-data/test-data-systematic.xml")
 
 
 def clean_resulting_tree(tree: etree._Element) -> str:
     cleaned_str = regex.sub("\n", "", etree.tostring(tree, encoding=str))
     return regex.sub(" +", " ", cleaned_str)
 
 
-class TestParseAndCleanXML(unittest.TestCase):
-    def test_parse_and_clean_xml_file_same_regardless_of_input(self) -> None:
-        test_data_systematic_tree = etree.parse("testdata/xml-data/test-data-systematic.xml")
-        from_file = parse_and_clean_xml_file("testdata/xml-data/test-data-systematic.xml")
-        from_tree = parse_and_clean_xml_file(test_data_systematic_tree)
-        cleaned_from_file = clean_resulting_tree(from_file)
-        cleaned_from_tree = clean_resulting_tree(from_tree)
-        self.assertEqual(
-            cleaned_from_file,
-            cleaned_from_tree,
-            msg="The output must be equal, regardless if the input is a path or parsed.",
-        )
-
-    def test_annotations_regions_links_before(self) -> None:
-        test_data_systematic_tree = etree.parse("testdata/xml-data/test-data-systematic.xml")
-        annotations_regions_links_before = [
-            e for e in test_data_systematic_tree.iter() if regex.search("annotation|region|link", str(e.tag))
-        ]
-        self.assertGreater(len(annotations_regions_links_before), 0)
-
-    def test_annotations_regions_links_after(self) -> None:
-        from_file = parse_and_clean_xml_file("testdata/xml-data/test-data-systematic.xml")
-        annotations_regions_links_after = [
-            e for e in from_file.iter() if regex.search("annotation|region|link", str(e.tag))
-        ]
-        self.assertEqual(
-            len(annotations_regions_links_after),
-            0,
-            msg="The tags <annotation>, <region>, and <link> must be transformed to their technically correct form "
-            '<resource restype="Annotation/Region/LinkObj">',
-        )
-
-    def test_comment_removal(self) -> None:
-        from_file = parse_and_clean_xml_file("testdata/xml-data/test-data-systematic.xml")
-        comments = [e for e in from_file.iter() if isinstance(e, etree._Comment)]
-        self.assertEqual(
-            len(comments),
-            0,
-            msg="properties that are commented out would break the the constructor of the class XMLProperty, "
-            "if they are not removed in the parsing process",
-        )
+def test_parse_and_clean_xml_file_same_regardless_of_input(
+    data_systematic_unclean: Any, data_systematic_cleaned: etree._Element
+) -> None:
+    from_tree = parse_and_clean_xml_file(data_systematic_unclean)
```

Comment:
Now, you mix 2 things:
 - cleaned versus uncleaned
 - from tree versus from file

For me it is now very unclear what exactly is tested, and what is not tested

Reply:
I did not change the content of the test, the difference was that I used the fixture. But I understand that when we want to test the parser function it is clearer if it is actually called, even if it is redundant. So I changed that.

Reply:
Yes, now it's clear. This might be the well-known exception from the DRY-rule :-)